### PR TITLE
Move halberd from Medieval Stuff mod

### PR DESCRIPTION
**Pull request by Rail-Runner**
Move halberd and its' fake version from Medieval Stuff mod into the main game.

Other changes:
1. Remove AXE quality from halberd.
2. Add COOK 1 quality to halberd. One could use a spike on its' end to cook a piece of meat over a fire.
3. Remove LUMBER action from halberd.
4. Add NON_STUCK flag to halberd.
5. Reduce spawn rate of halberd in some item groups.
6. Add halberd and its' fake version to No Medieval Items mod's blacklist.
7. Remove unused BASH flag from some items.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
> Increaxe AXE quality of halberd from 1 to 2.

AXE is for wood chopping. Halberd would make a terrible wood chopping tool due to how thin and light it is. A kukri would be more useful for chopping wood than a halberd and kukri doesn't have any AXE quality.

> Add IMPALE technique to halberd.

Nailing 2 zombies with one thrust of the halberd would require the whole axe blade part to go through the first one.

> Add REACH3 flag to halberd and its' fake version.

REACH3 is for pikes, not spears. Halberd is much shorter than a pike.

**Comment by chaosvolt** [comment]
Swiggith swooty.

Yeah, I picked wood-cutting of 1 for the halberd because I figured it would be inferious as a tool to the battleaxe, which already has 2 quality. Giving it a cooking quality wasn't something I'd thought of, though. Clever.

And yeah, Coolthulhu mentioned the oddness of giving it impale back when I first added the halberd. X3

**Comment by Malkeus** [comment]
Typical Spear: http://bashapedia.pbworks.com/f/1294528734/greek_armour__spear_tip.jpg
Typical Halberd: http://www.coldsteel-uk.com/store/maa-swiss-halberd-89msw-full-1.jpg

The shape would pretty much preclude impaling, much like the cross bar on a boar spear. I know a hydraulic muscled large mutant with very strong could probably do it, but we don't have a way to conditionally impale, do we? The aforementioned mutant could probably impale a zombie with the flat side of a door also...so, yeah...

**Comment by kevingranade** [comment]
Something else about impaling, the Halbert should have the flag (forgot the
name) that prevents sticking, impaling weapons should not have that flag.

**Comment by chaosvolt** [comment]
Hmm. Might be acceptable, sure. I think it was `NON_STUCK` if I recall?

**Comment by Rivet-the-Zombie** [comment]
@chaosvolt, you are correct.

**Comment by Coolthulhu** [comment]
Non-stuck prevents also chopping kind of sticking, though. Like what axes and machetes get.

**Comment by chaosvolt** [comment]
Hmm, true. Also, if I recall, I asked whether giving it both CHOP and SPEAR would have any interesting effects back when, and the answer was no.  Not sure if there's some way to represent it being capable of both.

File: data/json/items/melee.json [hunk 1]
```diff
@@ -2018,6 +2018,40 @@
...
+  },{
+        "id": "halberd_fake",
+        "type": "GENERIC",
+        "symbol": "/",
+        "color": "light_gray",
+        "name": "halberd",
+        "description": "This is a dull, cheaply made replica of a polearm with an axe blade, a spike, and other fun things attached to a long stick.",
+        "price": 5000,
+        "material": ["wood", "aluminum"],
+        "flags": ["BASH", "REACH_ATTACK"],
```
Comment (Coolthulhu) [inline data/json/items/melee.json]: There is no "BASH" flag.
Flags like "CHOP" and "SLICE" don't change the damage type, they only affect sticking chance.
Flags "SPEAR" and "STAB" do change the cutting type to piercing one.

Comment (Darkyhard) [inline data/json/items/melee.json]: Looks like this flag was present on a few more weapons. I removed them from there too.

Comment (chaosvolt) [inline data/json/items/melee.json]: Strange, I'm not sure where I got the "BASH" flag for replicas.

File: data/mods/No_Medieval_Items/modinfo.json [hunk 2]
```diff
@@ -40,6 +40,8 @@
         "scythe_war",
         "pike",
         "glaive",
+        "halberd",
+        "halberd_fake"
```
Comment (Coolthulhu) [inline data/mods/No_Medieval_Items/modinfo.json]: Missing separator

**Comment by kevingranade** [comment]
For my idea for nicer handling of the halberd, see #15034

**Comment by chaosvolt** [comment]
Maybe, but a way of choosing the desired attack type would be better. It would be odd to lose chopping in melee range.

**Pull request merged**